packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -224,6 +229,25 @@ function DataViews< Item >( {
 		};
 	}, [ infiniteScrollHandler, view.infiniteScrollEnabled ] );
 
+	// Filter out DataViewsPicker layouts.
+	const defaultLayouts = useMemo(
+		() =>
+			Object.fromEntries(
+				Object.entries( defaultLayoutsProperty ).filter(
+					( [ layoutType ] ) => {
+						return dataViewsLayouts.some(
+							( viewLayout ) => viewLayout.type === layoutType
+						);
+					}
+				)
+			),
+		[ defaultLayoutsProperty ]
+	);
+
+	if ( ! defaultLayouts[ view.type ] ) {
+		return null;
+	}
+
```

Comment:
I checked this bit, and it prevents this bug from happening (note how the active layout's icon is table even though it wasn't part of the default layouts):

https://github.com/user-attachments/assets/c5d05248-fb4d-4966-888d-f60dd9cbcb66

The other two cases (no prop: throws error) and the normal flow (view.type matches one of the defaultLayouts) don't change. This code will need to change when we allow registering 3rd party layouts, but the fact that is centralized will help to implement it.

Reply:
We could also handle it differently: instead of not rendering anything, provide a "no layout defined", or loading one of the default layouts, etc. But this is not something a user should ever have to deal with, so it's fine iterating when/if need it.

Reply:
And providing defaults so that `defaultLayouts` is not mandatory would also help devexp, specially for the picker that only has one layout (no need to do it here).